contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -742,51 +768,152 @@ have a built-in `nil` value.
+ 3. controllers: arbitrary changes (within the bounds of what is allowed) can
+    be made to a resource after the API call has completed.
+
+Some care is required when deciding which mechanism to use and managing the
+semantics.
+
+### Static Defaults
+
+Static default values are specific to each API version.  The default field
+values applied when creating an object with the "v1" API may be different than
+the values applied when using the "v2" API.  In most cases, these values are
+defined as literal values by the API version (e.g. "if this field is not
+specified it defaults to 0").  In some cases, these values may be
+deterministically derived from other fields (e.g. "if this field is not
+specified it defaults to the value of otherField").  In very rare cases, these
+values may be allocated from some pool or determined by some other method (e.g.
+Service's IP and IP-family related fields).
+
+These values are applied synchronously by the API server during the conversion
+from API-versioned declarative configuration to internal objects representing
+the desired state (`Spec`) of the resource. When the API call completes, all
+static defaults will have been set.  Subsequent GETs of the resource will
+include the default values explicitly.
+
+Static defaults are the best choice for values which are logically required,
+but which have a value that works well for most users.  Static defaulting
+must not consider any state except the object being operated upon (and the
+complexity of Service API stands as an example of why).
 
 Default values can be specified on a field using the `+default=` tag. Primitives
 will have their values directly assigned while structs will go through the
 JSON unmarshalling process. Fields that do not have an `omitempty` json tag will
 default to the zero value of their corresponding type if no default is assigned.
 
-Refer to [defaulting docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting) for more information.
+Refer to [defaulting docs](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting)
+for more information.
+
+### Admission Controlled Defaults
+
+In some cases, it is useful to set a default value which is not derived from
+the object in question.  For example, when creating a PersistentVolumeClaim,
+the storage class must be specified.  For many users, the best answer is
+"whatever the cluster admin has decided for the default".  StorageClass is a
+different API than PersistentVolumeClaim, and which one is denoted as the
+default may change at any time.  Thus this is not eligible for static
+defaulting.
 
-API version-specific default values are set by the API server.
+Instead, we can provide a built-in admission controller or a
+MutatingWebhookConfiguration.  Unlike static defaults, these may consider
+external state (such as annotations on StorageClass objects) when deciding
```

Comment:
do we want to add a note mentioning that this cab make them racy?

We have the example of the default class, there is always a time since a new class is created and the admission controller sees it, a new request can be received during that time ...
 see the multip

Reply:
ACK